Library/Homebrew/brew.sh

```diff
@@ -758,7 +758,7 @@ then
   export HOMEBREW_DEVELOPER_MODE="1"
 fi
 
-if [[ -n "${HOMEBREW_INSTALL_FROM_API}" && -n "${HOMEBREW_DEVELOPER_COMMAND}" ]]
+if [[ -n "${HOMEBREW_INSTALL_FROM_API}" && -n "${HOMEBREW_DEVELOPER_COMMAND}" && "${HOMEBREW_COMMAND}" != "irb" ]]
```

Comment:
Wondering if we should turn this into a more explicit denylist or just remove this entirely?

Reply:
Could probably remove this, since I want to allow `brew ruby` too. @Rylan12 will have a better idea about this.

Reply:
Yeah, there's no reason to exclude `HOMEBREW_INSTALL_FROM_API` from devs for those commands (except maybe `brew style formula` and `unpack`), it's just easier to maintain that developers shouldn't use it. Otherwise, we need to maintain this list to make sure that if commands get an online component they are removed from the list (and vice versa).

Another, better option, is probably just to complain on a per-command basis. We can remove the restriction on developers and just fail immediately in `brew bump-formula-pr` and friends if the user has `HOMEBREW_INSTALL_FROM_API`.

Reply:
`brew style formula` could still work for non-core formulae and `brew unpack` could still be useful to unpack even without a local formula IMO.

> We can remove the restriction on developers and just fail immediately in brew bump-formula-pr and friends if the user has HOMEBREW_INSTALL_FROM_API.

This would work for me 👍🏻